src/validator/YieldValidator.cpp

```diff
@@ -162,6 +179,11 @@ Status YieldValidator::validateYieldAndBuildOutputs(const YieldClause *clause) {
 
         NG_RETURN_IF_ERROR(makeOutputColumn(column->clone().release()));
     }
+
+    if (!hasInputVar) {
+        // generate constant expression result into querycontext
+        genConstantExprValues();
+    }
```

Comment:
I think we maybe need to check whether `yield->where()` expression is a constant expression, if so we can control the filter node what rows to output. Here `hasInputVar` only checks yield columns, not include where condition expression. please consider this situation: `yield 1 as id | yield 'hello' where $-.id > 0`

Reply:
good catch